File: mybatis/src/main/java/data/spring/mybatis/adapter/in/product/ProductController.java
package data.spring.mybatis.adapter.in.product;

import data.spring.mybatis.adapter.in.product.dto.ProductUpdateBatchRequest;
import data.spring.mybatis.application.required.product.ProductUseCase;
import jakarta.validation.Valid;
import org.springframework.web.bind.annotation.*;

@RequestMapping("/products/v1")
@RestController
public class ProductController {
    private final ProductUseCase productUseCase;

    public ProductController(ProductUseCase productUseCase) {
        this.productUseCase = productUseCase;
    }

    @GetMapping("/hello")
    public String hello() {
        return "Hello World!";
    }

    @PatchMapping("/update")
    public void updateProducts(@Valid @RequestBody ProductUpdateBatchRequest updateBatchRequest) {
        this.productUseCase.updateAll(updateBatchRequest.toCommands());
    }
}

File: mybatis/src/main/java/data/spring/mybatis/adapter/in/product/ProductController.kt
package data.spring.mybatis.adapter.`in`.product

import data.spring.mybatis.adapter.`in`.product.dto.ProductUpdateBatchRequest
import data.spring.mybatis.application.provided.product.ProductUseCase
import jakarta.validation.Valid
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PatchMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RequestMapping("/products/v1")
@RestController
class ProductController(
    val productUseCase: ProductUseCase
) {
    @GetMapping("/hello")
    fun hello(): String {
        return "Hello World!"
    }

    @PatchMapping("/update")
    fun updateProducts(@Valid @RequestBody updateBatchRequest: ProductUpdateBatchRequest) {
        this.productUseCase.updateAll(updateBatchRequest.toCommands())
    }
}